### PR TITLE
s4784738 - ADNI Image Classifier

### PR DESCRIPTION
**Pull request by Beenalf**
Hi COMP3710 teaching team, this is my submission to the pattern analysis repo. It classifies the brain images from the ADNI dataset using a GFNet model. Relevant results are printed and/or saved when training is run. The best accuracy I was able to achieve was 73% on the test set.

Files:
- README.md: Contains a description of the project and how it works
- dataset.py: Creates dataloaders using the ADNI dataset
- modules.py: Contains a GFNet model architecture
- train.py: Trains a model on the ADNI brain image dataset
- predict.py: Predicts the classification of a single image
- utils.py: Contains configuration settings for running training and predictions

Thanks for all the support this semester!
Benjamin Thatcher
s4784738

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Well-organized files.

**Problem Solving:**

- The algorithm solves the problem appropriately. 
- Accuracy in testing dataset: 0.73. Training and validation plots are shown in the report, but no evidence for the test acc: neither confusion matrix nor acc results.

**Model and functions:**

- It correctly uses PyTorch to construct the GFNet models and functions.
- With data augmentation.

**Code design:** Good.

**Code comment and docstring:**

- Good code comments 
- Good function docstrings

**Difficulty:** Hard.

**Additional Comments:**

- Insufficient introduction of GFNet, suspected ChatGPT style and content.
- The plots of training and val show that the model has been trained 38-39 epochs, while the hyperparameter in the code is set to 30 epochs, which doesn't match each other.
- There is no evidence to support the results in the report.
- There should be a conclusion in the report.
- Reasonable commits

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
The pull request includes a clear description about the file structure .
No previous feedback incorporated . 

File Organizing: Well-organized files.

Commit Log:
Commit messages are  progressive for the Recognition Problem solved using 4 files . 
Commits should  demonstrate a  more logical development flow.
Commits are regularly made, showing incremental development rather than bulk updates.

Documentation:
Code comments and docstrings  are included.
Proper GitHub markdown formatting is used, with organized headings, lists, and code blocks.
Can add about Conclusion and  future improvements in the ReadMe.

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>Header blocks. </td><td></td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Code and rport inconsistant.</td><td>-1</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting present. </td><td></td></tr>
	<tr><td></td><td>No Data leakage found. Uses Data Aug</td><td></td></tr>
	<tr><td></td><td>Difficulty : Hard.  GFNet (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Good Meaningful commit messages. </td><td></td></tr>
	<tr><td></td><td>Some/Adequate Progressive commits. Lacks flow.</td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. </td><td>-1</td></tr>
	<tr><td></td><td>Model/technical explanation :Acceptable. </td><td>-2</td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is adequate. </td><td>-1</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-6</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by shakes76** [comment]
Code and report inconsistent. -2 additonally.
